registrator_on_ec2_hostname.sh

```diff
@@ -1,4 +1,19 @@
-#!/bin/sh
-export LOCAL_IP=$(curl -s http://169.254.169.254/latest/meta-data/local-ipv4)
-echo Running: registrator -ip $LOCAL_IP $@ consul://$LOCAL_IP:8500
-exec registrator -ip $LOCAL_IP "$@" consul://$LOCAL_IP:8500
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+IFS=$'\n\t'
+
+function deregister {
+  registrator_exit_code=$?
+  echo "Registrator has exited; cleaning up containers registered on this instance" > /dev/stderr
+  set -o xtrace
+  curl --silent "http://${LOCAL_IP}:8500/v1/agent/services" | \
+    jq -r "map(\"http://${LOCAL_IP}:8500/v1/agent/service/deregister/\(.ID)\") | .[]" | \
+    xargs curl --silent --verbose -XPUT
```

Comment:
Using `xargs` like this should minimize the number of separate `curl` commands, letting `curl` reuse the TCP connection to Consul, without risking passing so many service URLs that the command line ends up being too long for the system to process.

Reply:
This is a very clever solution. Well done 👍